pilopt/tests/optimizer.rs

```diff
@@ -465,3 +465,39 @@ fn equal_constrained_transitive() {
     let optimized = optimize(analyze_string::<GoldilocksField>(input).unwrap()).to_string();
     assert_eq!(optimized, expectation);
 }
+#[test]
+fn simplify_associative_operations() {
+    let input = r#"namespace N(150);
+        col witness x;
+        col witness y;
+        col witness z;
+        col fixed c1 = [1]*;
+        col fixed c2 = [2]*;
+        col fixed c3 = [3]*;
+        
+        (x + c2) + c1 = y;
+        (c2 + x) + c3 = y;
+        (x - c2) + c1 = y;
+        
+        ((x + 3) - y) - 9 = z;
+        (c3 + (x + 3)) - y = z;
+        ((-x + 3) + y) + 9 = z;
+        ((-x + 3) + c3) + 12 = z;
+    "#;
+
+    let expectation = r#"namespace N(150);
+    col witness x;
+    col witness y;
+    col witness z;
+    N::x + 3 = N::y;
+    N::x + 5 = N::y;
+    N::x - 2 + 1 = N::y;
```

Comment:
Nice! I think as a follow-up PR, we should also try to simplify this and the next line.